atecc608a_se.c

```diff
@@ -240,17 +247,29 @@ static psa_status_t atecc608a_generate_key(psa_key_slot_number_t key_slot,
         return PSA_ERROR_NOT_SUPPORTED;
     }
 
-    if (p_pubkey_out != NULL && pubkey_out_size < ATCA_PUB_KEY_SIZE)
+    if (p_pubkey_out != NULL && pubkey_out_size < key_data_len)
```

Comment:
Actually you're going to write `1 + ATCA_PUB_KEY_SIZE` bytes, not some amount calculated from PSA macros. So the correct test here would be `pubkey_out_size < 1 + ATCA_PUB_KEY_SIZE`.

Reply:
I believe that that's the way I wrote it in my first approach. Do you agree for that, @Patater?

Reply:
For consistency, when writing to a buffer used by the PSA Crypto core, I'd recommend using sizes related to the PSA Crypto core formats. When writing to a buffer used by the driver, I'd recommend using driver sizes. This should help avoid using incorrect sizes. We also don't need to repeat the knowledge that `1 + ATCA_PUB_KEY_SIZE` is how to convert the size everywhere.

This is a stylistic choice, however. I'm happy with either style if we assert that we've written the correct number of bytes for the PSA format in a robust enough way that if the PSA format changes or the underlying driver format changes, we realize in an obvious way we need to update the format translation code.

Reply:
Here, the core guarantees that the buffer is at least `PSA_KEY_EXPORT_MAX_SIZE(PSA_KEY_TYPE_ECC_PUBLIC_KEY(PSA_ECC_CURVE_SECP256R1), 256)` bytes long (by the specification of the export functions), and that this is ≥65 (by the specification of the size macro). The check that's useful in the driver is that the buffer size is at least the number of bytes that the driver actually writes, and the driver writes 1 byte at offset 0 and `ATCA_PUB_KEY_SIZE` bytes at offset 1.